src/Entity/QuickNodeCloneEntityFormBuilder.php

```diff
@@ -26,39 +87,51 @@ public function getForm(EntityInterface $original_entity, $operation = 'default'
     foreach ($new_node->getTranslationLanguages() as $langcode => $language) {
       $translated_node = $new_node->getTranslation($langcode);
 
+      // Unset excluded fields.
+      if ($excludeFields = $this->getConfigSettings($translated_node->getType())) {
+        foreach($excludeFields as $key => $excludeField) {
+          unset($translated_node->{$excludeField});
+        }
+      }
+
       foreach ($translated_node->getFieldDefinitions() as $field_definition) {
         $field_storage_definition = $field_definition->getFieldStorageDefinition();
         $field_settings = $field_storage_definition->getSettings();
         $field_name = $field_storage_definition->getName();
-        if (isset($field_settings['target_type']) && $field_settings['target_type'] == "paragraph") {
+        if (isset($field_settings['target_type']) && $field_settings['target_type'] == 'paragraph') {
 
           // Each paragraph entity will be duplicated, so we won't be editing the same as the parent in every clone.
           if (!$translated_node->get($field_name)->isEmpty()) {
             foreach ($translated_node->get($field_name) as $value) {
               if ($value->entity) {
                 $value->entity = $value->entity->createDuplicate();
-                foreach($value->entity->getFieldDefinitions() as $field_definition) {
+                foreach ($value->entity->getFieldDefinitions() as $field_definition) {
                   $field_storage_definition = $field_definition->getFieldStorageDefinition();
                   $pfield_settings = $field_storage_definition->getSettings();
                   $pfield_name = $field_storage_definition->getName();
-                  \Drupal::moduleHandler()->alter('cloned_node_paragraph_field', $value->entity, $pfield_name, $pfield_settings);
+
+                  // Check whether this field is excluded and if so unset.
+                  if ($this->excludeParagraphField($pfield_name)) {
```

Comment:
We should be checking against the paragraph bundle too in case a field is used on multiple paragraph types. Perhaps we can pass the bundle as an arguiment here?

Reply:
I'm not sure how we can do this because we are not specifying which paragraph bundle the excluded field belongs to. This could be achieved in the more generic solution however.

Reply:
Resolved.